File: fibonacci/fib.lua
local function fib(n) 
	return n<2 and n or fib(n-1)+fib(n-2) 
end


local function main()
	local n = tonumber(arg[1])
	io.stdout:write("LANGUAGE LUA:" .. fib(n) .. "\n")
end
main()

File: fibonacci/fib.java
public class fib{
    private static long fib(long n){
        if(n < 2){
            return n;
        }
        return fib(n-1) + fib(n-2);
        
    }
    public static void main(String[] args){
        int n = Integer.parseInt(args[0]);
        System.out.println("LANGUAGE  JAVA: "+ fib(n));
    }
}